addons/missile_defense/functions/fnc_interceptorPFH.sqf

```diff
@@ -0,0 +1,66 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Brett Mayson
+ * Processes each interceptor
+ *
+ * Arguments:
+ * 0: Args <ARRAY>
+ * 1: Handle <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [ace_missile_defense_interceptorPFH] call CBA_fnc_addPerFrameHandler
+ *
+ * Public: No
+ */
+
+{
+    private _interceptors = _y getOrDefault ["interceptors", []];
+    {
+        _x params ["_projectile", "_target", "_lastPosition", "_lastDistance"];
+        if (isNull _projectile) then {
+            _interceptors deleteAt _forEachIndex;
+            if !(isNull _target) then {
+                (_y getOrDefault ["targets_tracking", []]) pushBack _target;
+                TRACE_2("Interceptor no longer exists",_projectile,_target);
+            };
+            continue;
+        };
+        if (isNull _target) then {
+            // TODO pick a different target?
+            _interceptors deleteAt _forEachIndex;
+            continue;
+        };
+        private _currentPosition = getPosASLVisual _projectile;
+        private _targetPosition = getPosASLVisual _target;
+
+        private _posDiff = (_currentPosition vectorDiff _lastPosition);
+        private _lengthSqr = _posDiff vectorDotProduct _posDiff;
+        private _minDistance = if (_lengthSqr - 0.001 <= 0) then {
+            _lastPosition vectorDistance _targetPosition
+        } else {
+            private _d = (_targetPosition vectorDiff _lastPosition) vectorDotProduct (_currentPosition vectorDiff _lastPosition);
+            private _t = 0 max (1 min (_d / _lengthSqr));
+            private _projection = _lastPosition vectorAdd ((_currentPosition vectorDiff _lastPosition) vectorMultiply _t);
+            _projection vectorDistance _targetPosition;
```

Comment:
not used anywhere

Reply:
It will be returned regardless, but it is good for visually representing that we will use the returned value